lib/place_picker.dart

```diff
@@ -23,9 +23,31 @@ class LocationResult {
   /// Formatted address suggested by Google
   String formattedAddress;
 
+  AddressComponent country;
+
+  AddressComponent city;
+
+  AddressComponent administrativeAreaLevel1;
+
+  AddressComponent administrativeAreaLevel2;
+
+  AddressComponent subLocalityLevel1;
+
+  AddressComponent subLocalityLevel2;
+
+  String postalCode;
+
   String placeId;
 }
 
+class AddressComponent {
+  String name;
+  String code;
```

Comment:
I'll suggest you use `shortName` instead. `code` feels too cryptic and will demand extra documentation to understand. `shortName` is self explanatory